pkg/proxy/routes.go

```diff
@@ -196,3 +207,26 @@ func removeFromProxyRoutesIPv6() error {
 
 	return nil
 }
+
+// removeStaleProxyRulesIPv6 removes stale proxy rules. This is a v1.15 only function.
+func removeStaleProxyRulesIPv6() error {
+	return removeProtoUnspecRules(netlink.FAMILY_V6)
+}
+
+// removeProtoUnspecRules removes all routing rules with protocol RTPROT_UNSPEC. This is a v1.15 only function.
+func removeProtoUnspecRules(family int) error {
+	rules, err := route.ListRules(family, nil)
+	if err != nil {
+		return fmt.Errorf("listing routing rules: %w", err)
+	}
+	for _, r := range rules {
+		if r.Protocol == unix.RTPROT_UNSPEC {
+			if err := netlink.RuleDel(&r); err != nil {
+				if !errors.Is(err, syscall.ENOENT) && !errors.Is(err, syscall.EAFNOSUPPORT) {
+					return fmt.Errorf("removing proto unspec routing rule: %w", err)
+				}
+			}
+		}
+	}
+	return nil
+}
```

Comment:
Would it help for review to split the custom cleanup logic into a separate patch (and make the first patch a clean cherry-pick from upstream) ? Or is it crucial for bisectability that all these changes go in as one patch?

Reply:
Seems no harm to split into two patches. 1.15-old -> 1.15-tip upgrade is the only one that could be affected in terms of git bisect, but I don't see too many risks.